### PR TITLE
Add checkbox component

### PR DESCRIPTION
**Pull request by ThatOtherPerson**

### CODE REVIEW

File: src/Checkbox/index.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    this.setState({ id });
+  }
+
+  render() {
+    const { checked, onChange, children } = this.props;
+
+    return (
+      <span className={styles.checkboxContainer}>
+        <input checked={checked} onChange={onChange} id={this.state.id} type="checkbox" {...getRestProps(this.props, propTypes)} />
+        <label htmlFor={this.state.id}>{children ? <span>{children}</span> : null}</label>
```
Comment (frankleng) [inline src/Checkbox/index.js]: @ThatOtherPerson how about ```children && <span>{children}</span>``` ?

File: src/Checkbox/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+const React = require('react');
+const { getRestProps } = require('../utils');
+const _ = require('lodash');
+
+const styles = require('./Checkbox.scss');
+
+const propTypes = {
+  id: React.PropTypes.string,
```
Comment (frankleng) [inline src/Checkbox/index.js]: @ThatOtherPerson shouldn't this be required?

Comment (ThatOtherPerson) [inline src/Checkbox/index.js]: Why? It's set up to automatically generate an id if one isn't provided.

Comment (frankleng) [inline src/Checkbox/index.js]: might be a bit harder to debug if the unique id is random and not semantic
your call. not a huge deal.

Comment (frankleng) [inline src/Checkbox/index.js]: @ThatOtherPerson 2nd thought. ids might collide if we make it semantic... nvm.

**Pull request merged**